Implement tsv tools

14thibea opened the pull request:
Tsv tools already exist in the code but  cannot be used with the general commandline.
The goal of this PR is to add the following tsv tools to the general commandline:
- **restriction**, to reproduce the restrictions applied to AIBL and OASIS datasets,
- **extract**, to extract the lists of participants for each diagnosis,
- **split**, to perform one split on extracted diagnoses (used in the original paper for train / test split),
- **kfold**, to perform a k-fold split on extracted diagnoses (used in the original paper for train / validation splits)
- **analysis**, to retrieve the demographics tables from the extracted diagnoses.

The commandline would look like this:
`clinicadl tsvtool {restriction|extract|split|kfold|analysis} <args>`

Do not hesitate to propose more appropriate names or descriptions for the tools !
I did not plan to include the `test` function which checks if there is data leakage in the splits. Please comment if you think it is needed and how it should be added.

pep8speaks commented:
Hello @14thibea! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`clinicadl/clinicadl/tools/tsv/restriction.py`](https://github.com/aramis-lab/AD-DL/blob/ef2a96fc6c6c0521c728433847c1b9f7c61aab01/clinicadl/clinicadl/tools/tsv/restriction.py):

> [Line 24:1](https://github.com/aramis-lab/AD-DL/blob/ef2a96fc6c6c0521c728433847c1b9f7c61aab01/clinicadl/clinicadl/tools/tsv/restriction.py#L24): [W391](https://duckduckgo.com/?q=pep8%20W391) blank line at end of file

14thibea commented:
I removed the intermediate folder _lists_by_diagnosis_ as this was useful when we had also a _lists_by_task_ organization but we abandonned it.

mdiazmel commented on clinicadl/clinicadl/tools/tsv/demographics_analysis.py:
```diff
@@ -9,51 +9,42 @@
+        raise ValueError("The column corresponding to mmse is not labelled as %s. "
+                         "Please change the parameters." % mmse_name)
 
     columns = ['n_subjects', 'mean_age', 'std_age', 'min_age', 'max_age', 'sexF', 'sexM', 'mean_MMSE', 'std_MMSE',
```
Shorter lines improves readability of code, e.g.:
```suggestion
    columns = ['n_subjects', 'mean_age', 'std_age',
               'min_age', 'max_age', 'sexF',
               'sexM', 'mean_MMSE', 'std_MMSE',
               'min_MMSE', 'max_MMSE', 'CDR_0',
               'CDR_0.5', 'CDR_1', 'CDR_2',
               'CDR_3', 'mean_scans', 'std_scans',
               'n_scans']
```

mdiazmel commented on clinicadl/clinicadl/tools/tsv/demographics_analysis.py:
```diff
@@ -9,51 +9,42 @@
+                         "Please change the parameters." % mmse_name)
 
     columns = ['n_subjects', 'mean_age', 'std_age', 'min_age', 'max_age', 'sexF', 'sexM', 'mean_MMSE', 'std_MMSE',
                'min_MMSE', 'max_MMSE', 'CDR_0', 'CDR_0.5', 'CDR_1', 'CDR_2', 'CDR_3', 'mean_scans', 'std_scans',
```
To remove if last suggestion is accepted.
```suggestion

```

mdiazmel commented on clinicadl/clinicadl/tools/tsv/demographics_analysis.py:
```diff
@@ -9,51 +9,42 @@
 
     columns = ['n_subjects', 'mean_age', 'std_age', 'min_age', 'max_age', 'sexF', 'sexM', 'mean_MMSE', 'std_MMSE',
                'min_MMSE', 'max_MMSE', 'CDR_0', 'CDR_0.5', 'CDR_1', 'CDR_2', 'CDR_3', 'mean_scans', 'std_scans',
                'n_scans']
```
To remove if last suggestion is accepted.
```suggestion

```

mdiazmel commented on clinicadl/clinicadl/tools/tsv/restriction.py:
```diff
@@ -0,0 +1,23 @@
+from copy import deepcopy
+
+
+def aibl_restriction(merged_tsv, results_path):
```
Add a short paragraph to describe shortly these restrictions on AIBL.

mdiazmel commented on clinicadl/clinicadl/tools/tsv/restriction.py:
```diff
@@ -0,0 +1,23 @@
+    results_df.to_csv(results_path, sep='\t')
+
+
+def oasis_restriction(merged_tsv, results_path):
```
Short description of the Oasis' restrictions.

14thibea commented on clinicadl/clinicadl/cli.py:
```diff
@@ -695,17 +754,173 @@ def parse_command_line():
+    tsv_restrict_subparser.set_defaults(func=tsv_restrict_func)
+
+    tsv_extract_subparser = tsv_subparser.add_parser(
+        'extract',
```
Changed in commit https://github.com/aramis-lab/AD-DL/pull/18/commits/fc883c36595395546c80bd01e75b82e51f1f3764

14thibea commented on README.md:
````diff
@@ -131,6 +136,24 @@ For detailed instructions and options of each task type  `clinica 'task' -h`.
+Typical use for `tsvtool extract`:
+
+```text
+clinicadl tsvtool extract <merged_tsv> <missing_mods> <results_path> --restriction_path <restriction_path>
````
resolved by commit https://github.com/aramis-lab/AD-DL/pull/18/commits/b9cacca8a0e622a92730bd75aec20bffee0405f9

14thibea commented:
Can you tell me if you are OK with the format of the docstring in the last commit https://github.com/aramis-lab/AD-DL/pull/18/commits/ce4435b7fd027d371bbb4207c12d36003f0b405e ? If not can you give me an example ?

14thibea commented:
The removal of `lists_by_diagnosis` only impacts the `.sh` files in `scripts`. I find it easier to use without (I always forgot it when I wrote a new launcher), however it is true that it can be useful to stay compliant with AD-ML standard, in which data is organized by diagnosis or tasks lists.

14thibea commented:
Actually I am currently adding some changes to the test suite to use it more easily (without adding it to the commandline) could you maybe wait until I finish this ?

mdiazmel commented:
> Actually I am currently adding some changes to the test suite to use it more easily (without adding it to the commandline) could you maybe wait until I finish this ?

Yes, no problem. Ping me here or let me know when the PR is ready to merge!

14thibea commented:
It's OK now, I added the test suite.
I also retested everything with the OASIS dataset and fixed some bugs accordingly (I tested everything with ADNI the first time).
